sdk/core/core-rest-pipeline/src/policies/bearerTokenChallengeAuthenticationPolicy.ts

```diff
@@ -0,0 +1,297 @@
+   */
+  getToken: (scopes: string | string[], options: GetTokenOptions) => Promise<AccessToken | null>;
+  /**
+   * Request that the policy is trying to fulfill.
+   */
+  request: PipelineRequest;
+  /**
+   * Response containing the challenge.
+   */
+  response?: PipelineResponse;
+  /**
+   * Function that allows easily assigning a token to the request.
+   */
+  setAuthorizationHeader: (token: string) => void;
+}
+
+/**
+ * Options to override the processing of [Continuous Access Evaluation](https://docs.microsoft.com/azure/active-directory/conditional-access/concept-continuous-access-evaluation) challenges.
+ */
+export interface ChallengeCallbacks {
+  /**
+   * Allows for the authentication of the main request of this policy before it's sent.
+   * The `setAuthorizationHeader` parameter received through the `ChallengeCallbackOptions`
+   * allows developers to easily assign a token to the ongoing request.
+   */
+  authorizeRequest?(options: ChallengeCallbackOptions): Promise<void>;
+  /**
+   * Allows to handle authentication challenges and to re-authorize the request.
+   * The response containing the challenge is `options.response`.
+   * The `setAuthorizationHeader` parameter received through the `ChallengeCallbackOptions`
+   * allows developers to easily assign a token to the ongoing request.
+   * If this method returns true, the underlying request will be sent once again.
+   */
+  authorizeRequestOnChallenge(options: ChallengeCallbackOptions): Promise<boolean>;
+}
+
+/**
+ * Options to configure the bearerTokenChallengeAuthenticationPolicy
+ */
+export interface BearerTokenChallengeAuthenticationPolicyOptions {
+  /**
+   * The TokenCredential implementation that can supply the bearer token.
+   */
+  credential: TokenCredential;
+  /**
+   * The scopes for which the bearer token applies.
+   */
+  scopes: string | string[];
+  /**
+   * Allows for the processing of [Continuous Access Evaluation](https://docs.microsoft.com/azure/active-directory/conditional-access/concept-continuous-access-evaluation) challenges.
+   * If provided, it must contain at least the `authorizeRequestOnChallenge` method.
+   * If provided, after a request is sent, if it has a challenge, it can be processed to re-send the original request with the relevant challenge information.
+   */
+  challengeCallbacks?: ChallengeCallbacks;
+}
+
+/**
+ * Retrieves a token from a token cache or a credential.
+ */
+export async function retrieveToken(
+  options: ChallengeCallbackOptions
+): Promise<AccessToken | undefined> {
+  const { scopes, claims, getToken, request } = options;
+
+  const getTokenOptions: GetTokenOptions = {
+    claims,
+    abortSignal: request.abortSignal,
+    tracingOptions: request.tracingOptions
+  };
+
+  return (await getToken(scopes, getTokenOptions)) || undefined;
+}
+
+/**
+ * Default authorize request
+ */
+export async function defaultAuthorizeRequest(options: ChallengeCallbackOptions): Promise<void> {
```

Comment:
another strange export... I feel like a policy should only export its factory method and whatever interfaces are inputs to said factory

Reply:
Looks a left-over when it is in `bearerTokenAuthenticationPolicy`.

Reply:
removed